From 30205c8268d0a82c499b129037be1741a21f95d8 Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Sat, 1 Nov 2025 21:32:44 +0100
Subject: [PATCH] Disable MOBILEAP_11AX support

Rebased for the S24U BP4A semwifi-service: isSupportedSoftAp11ax() is
public there, so call it directly instead of through a bridge accessor.

---
diff --git a/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali b/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali
index f17a0a0..79d0d11 100644
--- a/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali
+++ b/smali/com/samsung/android/server/wifi/ap/SemSoftApConfiguration.smali
@@ -2028,7 +2028,7 @@
 
     invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
 
-    const-string v1, "SPF_Enhanced=true\nSPF_Enhanced_Menu=d2d\nSPF_Guest=lite\nSPF_MobileAp=true\nSPF_Concurrency=true\nSPF_11ax=true\nSPF_6G=true\nSPF_Dongle=false\nSPF_DualAp=true\n"
+    const-string v1, "SPF_Enhanced=true\nSPF_Enhanced_Menu=d2d\nSPF_Guest=lite\nSPF_MobileAp=true\nSPF_Concurrency=true\nSPF_11ax=false\nSPF_6G=true\nSPF_Dongle=false\nSPF_DualAp=true\n"
 
     invoke-virtual {v0, v1}, Ljava/lang/StringBuffer;->append(Ljava/lang/String;)Ljava/lang/StringBuffer;
 
diff --git a/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali b/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali
index 2f99a33..22a4dad 100644
--- a/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali
+++ b/smali/com/samsung/android/server/wifi/ap/SemWifiApChipInfo.smali
@@ -1714,11 +1714,11 @@
 .end method
 
 .method public supportThisSoftApFeature(I)Z
-    .locals 3
+    .locals 2
 
     const/4 v0, 0x1
 
-    if-eq p1, v0, :cond_2
+    if-eq p1, v0, :cond_1
 
     const/4 v1, 0x2
 
@@ -1726,11 +1726,9 @@
 
     const/4 v1, 0x3
 
-    const/4 v2, 0x0
-
     if-eq p1, v1, :cond_0
 
-    return v2
+    goto :goto_1
 
     :cond_0
     iget-boolean p1, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSupportPowerSave:Z
@@ -1739,22 +1737,36 @@
 
     iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSemWifiApChipSupportInfo:Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;
 
-    if-eqz p0, :cond_1
+    if-eqz p0, :cond_3
 
     invoke-virtual {p0}, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;->isSupportedSoftApPowerSave()Z
 
     move-result p0
 
-    if-eqz p0, :cond_1
+    if-eqz p0, :cond_3
 
     goto :goto_0
 
     :cond_1
-    return v2
+    iget-object p0, p0, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo;->mSemWifiApChipSupportInfo:Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;
+
+    if-eqz p0, :cond_3
+
+    invoke-virtual {p0}, Lcom/samsung/android/server/wifi/ap/SemWifiApChipInfo$SemWifiApChipSupportInfo;->isSupportedSoftAp11ax()Z
+
+    move-result p0
+
+    if-eqz p0, :cond_3
 
     :cond_2
     :goto_0
     return v0
+
+    :cond_3
+    :goto_1
+    const/4 p0, 0x0
+
+    return p0
 .end method
 
 .method public supportWifiSharing()Z
-- 
2.55.0
